### DISCUSSION TITLE
Installation on Raspberry Pi 4

### DISCUSSION THREAD

**Discussion by b-onc** [Freika/dawarich#302] [Q&A] [answered]
I'm trying to install dawarich on Raspberry Pi 4 using Portainer. I get this error:
```
Failed to deploy a stack: dawarich_sidekiq Pulling dawarich_db Pulling dawarich_redis Pulling dawarich_app Pulling no matching manifest for linux/arm/v7 in the manifest list entries
```

**Comment by mattisaoldgit** [comment]
Hi, I am not a Dawarich expert, just a fan! 

I had a look at the OS tags in the dockerhub and I could only see linux/ams64, which as far as I know means that it will not work on a Pi.  [Docker Hub Link](https://hub.docker.com/r/freikin/dawarich/tags)

I am sure you could build it and run it or maybe @Freika has already done it or some other magic :)

Matt

**Reply by Freika** [reply]
There are both AMD64 and ARM64 images so it should work on Pi as well 🤔 
![изображение](https://github.com/user-attachments/assets/e2e1c2a5-c94a-4326-a676-b1b85498e286)

**Reply by gboudreau** [reply]
@Freika The RPi4 uses the `arm/v7` architecture.
@b-onc Maybe you could try to git clone the repo, and use `docker build` on your RPi to create `arm/v7` versions of each of the images?

**Reply by b-onc** [reply]
@Freika you're right, I use photoprism and it correctly installed `arm64` version.
@gboudreau I'm fighting this for 2 days now and I'm really confused... `uname -a` reads:
> Linux raspberrypi4 6.1.21-v8+ #1642 SMP PREEMPT Mon Apr  3 17:24:16 BST 2023 aarch64 GNU/Linux

so it seems I have arm64?

I have the same issue with heimdall and asked a question there too: https://github.com/linuxserver/Heimdall/discussions/1378

I messed up somewhere in my journey but I can't figure out where...

update:
@gboudreau RPi4 has arm64 cpu but arm/v7 kernel, [source](https://raspberrypi.stackexchange.com/questions/101215/why-raspberry-pi-4b-claims-that-its-processor-is-armv7l-when-in-official-specif):

> There are two pre-compiled kernels distributed with Raspbian, an ARMv6 one for the model 1's and Zeros, and an ARMv7 one for the multicore models. The latter works on the 64-bit ARMv8 models as well,1 since they are backward compatible.

Raspian was released with 64 bit kernel (I guess?) some time ago, so I guess I need to update my raspi OS...

**Reply by Freika** [reply]
I'll try to add arm64v7 to list of images that are being built and published on dockerhub

**Reply by Freika** [reply]
Added in https://github.com/Freika/dawarich/releases/tag/0.15.3, please try again

**Reply by Freika** [reply, marked as answer]
Yeah, should work now: 
<img width="243" alt="изображение" src="https://github.com/user-attachments/assets/2daf5922-9ce8-4599-a577-1a6db9128a54">

**Comment by b-onc** [comment]
@Freika Thank you, it's all smooth now!

Well except one thing: I can't connect to dawarich using my local domain name.

I can connect using `192.168.178.78:3003`, but I can't using `raspberrypi4.local:3003`
Other web services such as homepage works correctly, both from `192.168.178.78:3000` and `raspberrypi4.local:3000`

I've changed 
```
APPLICATION_HOST: localhost
APPLICATION_HOSTS: localhost
```
to
```
APPLICATION_HOST: 0.0.0.0
APPLICATION_HOSTS: 0.0.0.0/0.0.0.0
```
but didn't work.

Here's what I see:
<img width="746" alt="image" src="https://github.com/user-attachments/assets/d3253d34-cfa5-493c-9cd9-24f05ec66ad7">

**Comment by Freika** [comment]
Add your host to `APPLICATION_HOSTS` as following:

`APPLICATION_HOSTS: 0.0.0.0,raspberry4.local`, without protocols, and ports aren't needed too (I think)

https://dawarich.app/docs/environment-variables-and-settings

**Reply by b-onc** [reply]
~Did not work :( So I tried with port (`APPLICATION_HOSTS: 0.0.0.0,raspberry4.local:3003`) and it did not work either~

Update: I copy/pasted your reply without realising you mistyped my local domain :D When I typed the correct domain (`raspberrypi4.local`, emphasis on `pi`) it worked, thank you!

**Reply by Freika** [reply]
Yeah, I do that a lot :) Glad it worked!
